src/hsmm_mvpy/models.py

```diff
@@ -464,16 +540,17 @@ def scale_parameters(self, eventprobs, n_events):
         params : ndarray
             shape and scale for the gamma distributions
         '''
-        averagepos = np.concatenate([np.arange(1,self.max_d+1)@eventprobs.mean(axis=1),
-                                     [self.mean_d]])#Durations
+        if eventprobs is not None:
+            averagepos = np.concatenate([np.arange(1,self.max_d+1)@eventprobs.mean(axis=1),
+                                         [self.mean_d]])#Durations
```

Comment:
I don't understand how you can take the mean of eventprobs to determine time. This will take the average of the probabilities. Shouldn't you do the dot product of samples and eventprobs?

Reply:
What I understand from what you suggest is what is being done in this line. First, take the mean of the probabilities by averaging across trials and then take the dotproduct with samples. Or do you mean to first do the dotproduct and then average?

Reply:
ah now I understand what you do. You can just directly do the dot product, it will directly take the average, no need to do that first.